src/tools/ScenarioMeasurement/Startup/Startup.cs

```diff
@@ -135,7 +137,22 @@ static void checkArg(string arg, string name)
 
         logger.Log($"Running {appExe} (args: \"{appArgs}\")");
         logger.Log($"Working Directory: {workingDir}");
-        if(runWithoutExit)
+
+        if (affinity > 0 && (OperatingSystem.IsWindows() || OperatingSystem.IsLinux()))
+        {
```

Comment:
Use variable for `Process.GetCurrentProcess()` and reuse it, instead of doing the call again.